kubespawner/reflector.py

```diff
@@ -372,17 +402,29 @@ def start(self):
         start of program initialization (when the singleton is being created),
         and not afterwards!
         """
-        if hasattr(self, 'watch_thread'):
-            raise ValueError('Thread watching for resources is already running')
+        if self.watch_task and not self.watch_task.done():
+            raise ValueError('Task watching for resources is already running')
 
-        self._list_and_update()
-        self.watch_thread = threading.Thread(target=self._watch_and_update)
-        # If the watch_thread is only thread left alive, exit app
-        self.watch_thread.daemon = True
-        self.watch_thread.start()
+        await self._list_and_update()
+        self.watch_task = asyncio.create_task(self._watch_and_update())
```

Comment:
`create_task` [requires Python 3.7](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task), but setup currently requires 3.6. No objection to bumping required Python in this PR since it's about time anyway, but the same result is accomplished with the older `asyncio.ensure_future` without changing the required Python.

Reply:
Let's bump requirements, since I've used create_task in a bunch of places.